packages/flutter/lib/src/foundation/binding.dart

```diff
@@ -40,6 +61,167 @@ abstract class BindingBase {
+  ///
+  /// This is used by development tools when the application code has
+  /// changed, to cause the application to pick up any changed code.
+  /// Bindings are expected to use this method to reregister anything
+  /// that uses closures, so that they do not keep pointing to old
+  /// code, and to flush any caches of previously computed values, in
+  /// case the new code would compute them differently.
+  void reassembleApplication() { }
+
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.name"), which takes no arguments and returns
+  /// no value.
+  ///
+  /// Invokes the `callback` callback when the service extension is
+  /// invoked.
+  void registerSignalServiceExtension({ 
+    @required String name,
+    @required VoidCallback callback
+  }) {
+    assert(name != null);
+    assert(callback != null);
+    registerServiceExtension(
+      name: name,
+      callback: (Map<String, String> parameters) async {
+        callback();
+        return <String, dynamic>{};
+      }
+    );
+  }
+
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.name"), which takes a single argument
+  /// "enabled" which can have the value "true" or the value "false"
+  /// or can be omitted to read the current value. (Any value other
+  /// than "true" is considered equivalent to "false". Other arguments
+  /// are ignored.)
+  ///
+  /// Invokes the `getter` callback to obtain the value when
+  /// responding to the service extension method being invoked.
+  ///
+  /// Invokes the `setter` callback with the new value when the
+  /// service extension method is invoked with a new value.
+  void registerBoolServiceExtension({ 
+    String name,
+    @required ValueGetter<bool> getter,
+    @required ValueChanged<bool> setter
+  }) {
+    assert(name != null);
+    assert(getter != null);
+    assert(setter != null);
+    registerServiceExtension(
+      name: name, 
+      callback: (Map<String, String> parameters) async {
+        if (parameters.containsKey('enabled'))
+          setter(parameters['enabled'] == 'true');
+        return <String, dynamic>{ 'enabled': getter() };
+      }
+    );
+  }
+
+  /// Registers a service extension method with the given name (full
+  /// name "ext.flutter.name"), which takes a single argument with the
+  /// same name as the method which, if present, must have a value
+  /// that can be parsed by [double.parse], and can be omitted to read
+  /// the current value. (Other arguments are ignored.)
+  ///
+  /// Invokes the `getter` callback to obtain the value when
+  /// responding to the service extension method being invoked.
+  ///
+  /// Invokes the `setter` callback with the new value when the
+  /// service extension method is invoked with a new value.
+  void registerNumericServiceExtension({ 
+    @required String name,
+    @required ValueGetter<double> getter,
+    @required ValueChanged<double> setter
```

Comment:
`ValueChanged` is a bit confusing as the effective value has not changed yet (at least for our current use-cases). Suggestion: `ValueSetter`, as it tell you that this callback is reponsible for setting that value.